R/get_v1_data.R

```diff
@@ -215,35 +265,159 @@ spod_get_zones_v1 <- function(
+  }
+  names(zones_sf)[names(zones_sf) == "ID"] <- "id"
+
+  # load and prepare id relations for districts
+  relations_districts <- readr::read_delim(
+    file = paste0(spod_get_data_dir(), "/",
+      spod_subfolder_raw_data_cache(1),
+      "relaciones_distrito_mitma.csv"),
+    delim = "|", show_col_types = FALSE
+  )
+  relations_districts_col_names <- names(relations_districts)
+  relations_districts_col_names <- gsub("distrito", "district", relations_districts_col_names)
+  relations_districts_col_names <- gsub("municipio", "municipality", relations_districts_col_names)
+  relations_districts_col_names <- gsub("^district$", "census_district", relations_districts_col_names)
+  names(relations_districts) <- relations_districts_col_names
+  
+  # load and prepare id relations for municipalities
+  relations_municipalities <- readr::read_delim(
+    file = paste0(spod_get_data_dir(), "/",
+      spod_subfolder_raw_data_cache(1),
+      "relaciones_municipio_mitma.csv"),
+    delim = "|", show_col_types = FALSE
+  )
+  relations_municipalities_col_names <- names(relations_municipalities)
+  relations_municipalities_col_names <- gsub("municipio", "municipality", relations_municipalities_col_names)
+  names(relations_municipalities) <- relations_municipalities_col_names
+
+  # summarise districts relations including municipality data
+  relations_districts_aggregated <- relations_districts |>
+    dplyr::left_join(
+      relations_municipalities |>
+        dplyr::group_by(.data$municipality_mitma) |>
+        dplyr::summarize(
+          municipalities = paste(.data$municipality, collapse = "; ")
+        ),
+      by = "municipality_mitma") |> 
+    dplyr::group_by(.data$district_mitma) |> 
+    dplyr::summarize(
+      census_districts = paste(.data$census_district, collapse = "; "),
+      municipalities_mitma = paste(.data$municipality_mitma, collapse = "; "),
+      municipalities = paste(.data$municipalities, collapse = "; ")
+    )
+
+  # summarise municipalities relations
+  relations_municipalities_aggregated <- relations_municipalities |>
+    dplyr::left_join(
+      relations_districts |> 
+        dplyr::group_by(.data$municipality_mitma) |>
+        dplyr::summarize(
+          census_districts = paste(.data$census_district, collapse = "; "),
+          districts_mitma = paste(.data$district_mitma, collapse = "; ")
+        )
+      , by = "municipality_mitma") |>
+    dplyr::group_by(.data$municipality_mitma) |> 
+    dplyr::summarize(
+      municipalities = paste(.data$municipality, collapse = "; "),
+      districts_mitma = paste(.data$districts_mitma, collapse = "; "),
+      census_districts = paste(.data$census_districts, collapse = "; ")
+    )
+  
+  # now we have some duplicate ids, we need to remove them
+  # here's a function for that
+  unique_separated_ids <- function(column) {
+    # Split the string by semicolon, remove duplicates, and join them back with semicolons
+    sapply(column, function(x) {
```

Comment:
`sapply` is not type-safe. Consider using `purrr` or `furrr` functions instead.

Reply:
will do